### PR TITLE
[#740] Refactor config apply

### PR DESCRIPTION
**Pull request by ljupcovangelski**
- [x] Refactor how to create/update configMaps with `airy config apply`
- [x] Change the env variables in the helm charts
- [x] Remove the configmaps for components which are not found in `airy.yaml`
- [x] Check and update the docs

Resolves: #740, #892, #1150

### CODE REVIEW

File: cli/pkg/cmd/config/config.go [hunk 1]
```diff
@@ -34,20 +36,33 @@ func applyConfig(cmd *cobra.Command, args []string) {
 		console.Exit("Could not find an installation of Airy Core. Get started here https://airy.co/docs/core/getting-started/installation/introduction")
 	}
 
-	if twilioApply(conf, clientset) {
-		fmt.Println("Twilio configuration applied.")
+	var configuredComponents []string
```
Comment (lucapette) [inline cli/pkg/cmd/config/config.go:39]: a map better fit for the code (so we don't loop and can get rid of the found function)

File: cli/pkg/cmd/config/config.go [hunk 2]
```diff
@@ -34,20 +36,33 @@ func applyConfig(cmd *cobra.Command, args []string) {
...
+	for componentType,_ := range conf.Components {
+		for componentName,componentValues := range conf.Components[componentType] {
+			configmapName := componentType+"-"+componentName
+			labels := map[string]string{
+				"core.airy.co/component": configmapName,
+			}		
+			applyErr := kube.ApplyConfigMap(configmapName, conf.Kubernetes.Namespace, componentValues, labels, clientset)
+			configuredComponents = append(configuredComponents,configmapName)
+			if applyErr != nil{
+				fmt.Printf("Unable to apply configuration for component: %s-%s\n Error:\n %v\n", componentType, componentName, applyErr)
```
Comment (lucapette) [inline cli/pkg/cmd/config/config.go:49]:
```suggestion
				fmt.Printf("unable to apply configuration for component: %s-%s\n Error:\n %v\n", componentType, componentName, applyErr)
```

File: cli/pkg/cmd/config/config.go [hunk 3]
```diff
@@ -34,20 +36,33 @@ func applyConfig(cmd *cobra.Command, args []string) {
...
+			} else {
+				fmt.Printf("Applied configuration for component: %s-%s\n", componentType, componentName)
```
Comment (lucapette) [inline cli/pkg/cmd/config/config.go:51]:
```suggestion
				fmt.Printf("applied configuration for component: %s-%s\n", componentType, componentName)
```

File: cli/pkg/kube/configmaps.go [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+}
+
+func DeleteConfigMap(configmapName string, namespace string, clientset *kubernetes.Clientset) error {
+	cm, _ := clientset.CoreV1().ConfigMaps(namespace).Get(context.TODO(), configmapName, v1.GetOptions{})
+	if cm.GetName() != "" {
+		err := clientset.CoreV1().ConfigMaps(namespace).Delete(context.TODO(),configmapName, v1.DeleteOptions{})
+		return err
+	} else {
+		return nil
+	}
```
Comment (lucapette) [inline cli/pkg/kube/configmaps.go:37]:
```suggestion
	}
	
	return nil

```

**Review by lucapette — COMMENTED** [review]
First pass!

File: infrastructure/helm-chart/charts/apps/charts/api/templates/security.yaml [hunk 5]
```diff
@@ -0,0 +1,9 @@
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: security
+  namespace: {{ .Values.global.namespace }}
+data:
+  jwtSecret: {{ .Values.jwtSecret | default (randAlphaNum 128) | quote }}
+  systemToken: {{ .Values.systemToken | default (randAlphaNum 24) | quote }}
+  allowedOrigins: {{ .Values.allowedOrigins | quote }}
```
Comment (ljupcovangelski) [inline infrastructure/helm-chart/charts/apps/charts/api/templates/security.yaml:9]: Once we want to have these values optional on `airy create` time, we can just remove them from here and create an empty configMap.

**Review by chrismatix — DISMISSED** [review]

**Review by chrismatix — APPROVED** [review]

**Pull request merged**